rvc/train/train.py

```diff
@@ -236,6 +236,13 @@ def run(
     if torch.cuda.is_available():
         torch.cuda.set_device(rank)
 
+    if torch.cuda.is_available() and torch.cuda.get_device_name("cuda:0").endswith("[ZLUDA]"):
```

Comment:
1. Merge with condition above.
2. Since current device is set, simply invoke `torch.cuda.get_device_name()`. It will get the current set device:
    ```suggestion
        if torch.cuda.get_device_name().endswith("[ZLUDA]"):
    ```